src/runtime/internal/scheduler.ts

```diff
@@ -52,17 +52,32 @@ export function add_flush_callback(fn) {
 const seen_callbacks = new Set();
 let flushidx = 0;  // Do *not* move this inside the flush() function
 export function flush() {
+	// Do not reenter flush while dirty components are updated, as this can
+	// result in an infinite loop. Instead, let the inner flush handle it.
+	// Reentrancy is ok afterwards for bindings etc.
+	if (flushidx !== 0) {
+		return;
+	}
+
 	const saved_component = current_component;
 
 	do {
 		// first, call beforeUpdate functions
 		// and update components
-		while (flushidx < dirty_components.length) {
-			const component = dirty_components[flushidx];
-			flushidx++;
-			set_current_component(component);
-			update(component.$$);
+		try {
+			while (flushidx < dirty_components.length) {
+				const component = dirty_components[flushidx];
+				flushidx++;
+				set_current_component(component);
+				update(component.$$);
+			}
+		} catch (e) {
+			// reset dirty state to not end up in a deadlocked state and then rethrow
+			dirty_components.length = 0;
+			flushidx = 0;
+			throw e;
```

Comment:
Can you tell an example that comes to the catch statement?
Is it intended to handle `Maximum call stack size exceeded`?

Is there any possibility that this may cause the unintended DOM structure and the runtime to go wrong behavior?

Reply:
There's on test that throws a runtime error (I believe some dev time check for svelte:element) and then all other tests afterwards fail because flush is in a broken state then. Yes, we could end up in a broken Dom state but at least the app might continue to work. The alternative would be to not do this but only do console.error in that check.